x-pack/plugins/lens/public/shared_components/coloring/color_to_value.tsx

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { euiLightVars, euiDarkVars } from '@kbn/ui-shared-deps/theme';
+import { isColorDark } from '@elastic/eui';
+import chroma from 'chroma-js';
+import { CustomPaletteState } from 'src/plugins/charts/public';
+
+// This module is dedicated to the palette to value mapping
+
+function findColorSegment(
+  value: number,
+  { min, max }: { min: number; max: number },
+  comparison: (value: number, bucket: number) => number,
+  colors: string[],
+  rangeMin?: number,
+  rangeMax?: number
+) {
+  const maxValue = rangeMax ?? max;
+  const minValue = rangeMin ?? min;
+  // assume uniform distribution within the provided range, can ignore stops
+  const step = (maxValue - minValue) / colors.length;
+
+  // what about values in range
+  const index = colors.findIndex((c, i) => comparison(value, minValue + (1 + i) * step) <= 0);
+  return colors[index] || colors[0];
+}
+
+function findColorsByStops(
+  value: number,
+  comparison: (value: number, bucket: number) => number,
+  colors: string[],
+  stops: number[]
+) {
+  const index = stops.findIndex((s) => comparison(value, s) < 0);
+  return colors[index] || colors[0];
+}
+
+function getNormalizedValueByRange(
+  value: number,
+  { range }: CustomPaletteState,
+  minMax: { min: number; max: number }
+) {
+  let result = value;
+  if (range === 'percent') {
+    result = (100 * (value - minMax.min)) / (minMax.max - minMax.min);
+  }
+  // for a range of 1 value the formulas above will divide by 0, so here's a safety guard
+  if (Number.isNaN(result)) {
+    return 1;
+  }
+  return result;
+}
+
+/**
+ * When stops are empty, it is assumed a predefined palette, so colors are distributed uniformly in the whole data range
+ * When stops are passed, then rangeMin/rangeMax are used as reference for user defined limits:
+ * continuity is defined over rangeMin/rangeMax, not these stops values (rangeMin/rangeMax are computed from user's stop inputs)
+ */
+export function workoutColorForValue(
```

Comment:
As mentioned in https://github.com/elastic/kibana/pull/95217#pullrequestreview-655281286 I think this should be part of the palette definition in the charts plugin. Is there a reason you didn't move it there?

Reply:
Sorry my bad. I moved them out to isolate and test them, but didn't move to the plugin yet.
Will update shortly with the move.